course/permissions.py

```diff
@@ -226,7 +226,7 @@ def is_object_visible(self, request, view, course_instance):
         return True
 
 
-class CourseModulePermission(MessageMixin, Permission):
+class CourseModulePermissionBase(MessageMixin, Permission):
```

Comment:
Does this mean that students are able to read hidden modules by using the API and JWT tokens? Normally, students can not see hidden modules nor any content in them.

Reply:
Right now, only teachers or graders/git manager may get a JWT access to a course, so students can't read them. The JWT read access is more akin to administrative read (e.g. the build logs in git manager) than to content read (e.g. course chapter files). I do not see a particular reason to expand the use of the tokens to students as well. Though, it might make sense to use the 'write' permission for all administrator-like purposes.

Reply:
OK, since no secret information is exposed to the students, this is good. :+1: